conan_provider.cmake

```diff
@@ -477,13 +477,27 @@ macro(conan_provide_dependency method package_name)
         construct_profile_argument(_host_profile_flags CONAN_HOST_PROFILE)
         construct_profile_argument(_build_profile_flags CONAN_BUILD_PROFILE)
         get_property(_MULTICONFIG_GENERATOR GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+        if(EXISTS "${CMAKE_SOURCE_DIR}/conanfile.py")
+            file(READ "${CMAKE_SOURCE_DIR}/conanfile.py" outfile)
+            if(NOT "${outfile}" MATCHES ".*CMakeDeps.*")
```

Comment:
I think this is potentially risky because `conanfile.py` is still Python, so `CMakeDeps` may be anywhere (in a comment, in an import, etc) and still not be properly instantiated.
On the other hand, if the string is not _ANYWHERE_ (as it is being checked here), then we know this is likely going to be an issue, so maybe the warning is OK

Reply:
Yes, it's what I noted in my comment. I still think it may be worth to keep it as it won't show whent it shouldn't and currently is not showing at all so this can cover most of the cases. But I could understand not wanting it because it's incosistent.